sdk/src/prices/price-module.ts

```diff
@@ -0,0 +1,245 @@
+import { AddressUtil } from "@orca-so/common-sdk";
+import { PublicKey } from "@solana/web3.js";
+import {
+  DecimalsMap,
+  defaultConfig,
+  defaultThresholdConfig,
+  PoolMap,
+  PriceMap,
+  TickArrayMap,
+} from ".";
+import { WhirlpoolContext } from "../context";
+import { PDAUtil, PoolUtil, SwapUtils } from "../utils/public";
+import { convertListToMap, filterNullObjects } from "../utils/txn-utils";
+import { calculatePricesForQuoteToken, convertAmount, isSubset } from "./calculate-pool-prices";
+
+export class PriceModule {
+  // fetchPoolPrices performs both the network requests and the price calculation logic
+  // If the caller already has the necessary data, they can use calculatePoolPrices
+  // If the caller has some of the data but not others, they can call individual fetch* functions
+  static async fetchPoolPrices(
+    ctx: WhirlpoolContext,
+    mints: PublicKey[],
+    config = defaultConfig,
+    thresholdConfig = defaultThresholdConfig,
+    refresh = true
+  ): Promise<PriceMap> {
+    const { poolMap, tickArrayMap, decimalsMap } = await PriceModule.fetchPriceCalculationData(
+      ctx,
+      mints,
+      config,
+      refresh
+    );
+
+    return PriceModule.calculatePoolPrices(
+      mints,
+      poolMap,
+      tickArrayMap,
+      decimalsMap,
+      config,
+      thresholdConfig
+    );
+  }
+
+  static async fetchPriceCalculationData(
+    ctx: WhirlpoolContext,
+    mints: PublicKey[],
+    config = defaultConfig,
+    refresh = true
+  ) {
+    const poolMap = await PriceModule.fetchPoolsForMints(ctx, mints, config, refresh);
+    const tickArrayMap = await PriceModule.fetchTickArraysForPools(ctx, poolMap, config, refresh);
+    const decimalsMap = await PriceModule.fetchDecimalsForMints(ctx, mints, refresh);
+
+    return { poolMap, tickArrayMap, decimalsMap };
+  }
+
+  static async fetchPoolsForMints(
```

Comment:
Slight concern about this loop now that we realize deriving PDA is pretty slow. We are generating PDAs for all combinations of mints & tick-spacing which may be overkill.

If we are using the price module in useTokenPrices, we will need these fetches to be performant. Suggest we add another fetch function here to allow fetching pools w/ address directly.